todo/views.py

```diff
@@ -1,6 +1,77 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
-from django.shortcuts import render
+from django.contrib.auth.forms import UserCreationForm
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpResponseRedirect
+from django.shortcuts import redirect
+from django.urls import reverse_lazy
+from django.views.generic import ListView, DetailView, FormView, DeleteView, CreateView
+
+from .forms import TaskForm
+from .models import Task
+
 
 # Create your views here.
+
+class AbstractLogin(LoginRequiredMixin):
+    login_url = '/login/'
+    redirect_field_name = reverse_lazy('tasklist')
+
+    class Meta:
+        abstract = True
+
+
+class TaskListView(AbstractLogin, ListView):
+    model = Task
+
+    def get_context_data(self, **kwargs):
+        context = super(TaskListView, self).get_context_data(**kwargs)
+        tasks = Task.objects.all()
+        filtered_tasks = [tasks.filter(status="todo"), tasks.filter(status="in_progress"), tasks.filter(status="done")]
```

Comment:
For sure it's better to pass it as dict, not a list, tasks.0 is not meaningful in template. 
context['todo_tasks']=
context['in_progress_tasks']=
context['done_tasks']=

Reply:
generally you're querying db 3 times here, I would do it that:
tasks_by_status = defaultdict(list)
for task in Task.objects.all():
    tasks_by_status[task.status] = task